meta-mentor-staging/recipes-core/meta/nativesdk-sdk-relocate.bb

```diff
@@ -35,6 +35,7 @@ sed -i -e "s#\$DEFAULT_INSTALL_DIR\>#\$installdir#g" "\$env_setup_script"
 END
     cat "${COREBASE}/meta/files/toolchain-shar-relocate.sh" >>sdk-relocate
     sed -i -e 's#grep -v "\$target_sdk_dir/.*"#grep -Ev "(sdk-relocate|$target_sdk_dir/(environment-setup-*|\.installpath))"#' sdk-relocate
+    sed -i -e '/native_sysroot=/a native_sysroot=$($SUDO_EXEC echo $native_sysroot | sed -e "s:\\$scriptdir:$target_sdk_dir:")' sdk-relocate
```

Comment:
Ah, it's due to the crappy way toolchain-shar-relocate.sh extracts OECORE_NATIVE_SYSROOT from the environment setup script. Since it doesn't expand the referenced variables, they get left as is. Well spotted, on this, thanks for your investigation. I never managed to reliably reproduce the problem in my setup for some reason.

Did you verify this works properly for both the SDK and ADE?

That said, I wonder if this would be better:
```
native_sysroot="$($SUDO_EXEC sh -c ". $env_setup_script && echo $OECORE_NATIVE_SYSROOT")"
```
That would let the shell expand any referenced variables, rather than using grep and cut to extract the value. Since we set relocate=1, we prevent any recursion resulting from environment-setup -> sdk-relocate -> environment-setup. Thoughts?